### PR TITLE
Added default Pantheon config for redis

### PR DESCRIPTION
**Pull request by johnennewdeeson**
Added default Redis config for D7 sites which was missing

### CODE REVIEW

**Review by mikeddeeson — COMMENTED** [review]
What if your site doesn't use redis or is not on Pantheon?

I think that if we are going to include this, then we should be checking for the environment var being pantheon at the very least

**Comment by angry-dan** [comment]
Yeah I agree - I wouldn't want a site using Redis when running on Docker if it's not using Redis in prod.

File: docroot/sites/all/conf/redis.settings.inc [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<?php
+
+/**
+ * @file
+ * Redis settings.
+ */
+
+$contrib_path = 'sites/all/modules/contrib';
+
+if (SETTINGS_ENVIRONMENT !== 'local') {
```
Comment (adam-dg) [inline docroot/sites/all/conf/redis.settings.inc]: We shouldn't be using the magic string "local" any more as we have predefined constants to use.

File: docroot/sites/all/conf/redis.settings.inc [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  $conf['redis_client_interface'] = 'PhpRedis';
+  $conf['cache_backends'][] = $contrib_path . '/redis/redis.autoload.inc';
+  $conf['cache_default_class'] = 'Redis_Cache';
+  $conf['cache_prefix'] = array('default' => 'pantheon-redis');
+  // Do not use Redis for cache_form (no performance difference).
+  $conf['cache_class_cache_form'] = 'DrupalDatabaseCache';
+  // Use Redis for Drupal locks (semaphore).
+  $conf['lock_inc'] = $contrib_path . '/redis/redis.lock.inc';
+}
+elseif (SETTINGS_INSTANCE === 'docker') {
```
Comment (adam-dg) [inline docroot/sites/all/conf/redis.settings.inc]: We shouldn't be using the magic string "local" any more as we have predefined constants to use.

**Comment by ghost** [comment]
So, some of these things are here for reference rather than 'YOU MUST HAVE IT IN YOUR PROJECT' - I'd just delete it if it didn't use Redis.

**Review by adam-dg — CHANGES_REQUESTED** [review]

File: docroot/sites/all/conf/redis.settings.inc (hunk 2 above)
Comment (ghost) [reply docroot/sites/all/conf/redis.settings.inc]: agreed

File: docroot/sites/all/conf/redis.settings.inc (hunk 1 above)
Comment (ghost) [reply docroot/sites/all/conf/redis.settings.inc]: agreed

**Pull request merged**